src/session/content/message_row/mod.rs

```diff
@@ -123,32 +125,41 @@ impl MessageRow {
     }
 
     fn show_delete_dialog(&self, revoke: bool) {
-        let window: Option<gtk::Window> = self.root().and_then(|root| root.downcast().ok());
+        let window: gtk::Window = self.root().and_then(|root| root.downcast().ok()).unwrap();
+
         let message = if revoke {
-            gettext("Do you want to delete this message for everyone?")
+            gettext("Do you want to delete this message for <b>everyone</b>?")
         } else {
             gettext("Do you want to delete this message?")
         };
-        let dialog = gtk::MessageDialog::new(
-            window.as_ref(),
-            gtk::DialogFlags::MODAL,
-            gtk::MessageType::Warning,
-            gtk::ButtonsType::YesNo,
-            &message,
-        );
-
-        dialog.run_async(clone!(@weak self as obj => move |dialog, response| {
-            if matches!(response, gtk::ResponseType::Yes) {
-                if let Ok(message) = obj.message().downcast::<Message>() {
-                    spawn(async move {
-                        if let Err(e) = message.delete(revoke).await {
-                            log::warn!("Error deleting a message (revoke = {}): {:?}", revoke, e);
-                        }
-                    });
+
+        let dialog = adw::MessageDialog::builder()
+            .heading(&gettext("Confirm Message Deletion"))
+            .body_use_markup(true)
+            .body(&message)
+            .modal(true)
+            .transient_for(&window)
+            .build();
+
+        dialog.add_responses(&[("no", &gettext("_No")), ("yes", &gettext("_Yes"))]);
+        dialog.set_default_response(Some("no"));
+        dialog.set_response_appearance("yes", adw::ResponseAppearance::Destructive);
+
+        dialog.connect_response(
```

Comment:
`run_async` still seems to exists, I'd rather still use that, it also makes us save the additional indentation.

Reply:
> it also makes us save the additional indentation.

wdym?

Reply:
Oh, nevermind, there's a new parameter in that function, so there will always be an extra indent because of rustfmt.

Reply:
Btw, can you also use `run_async` in the other places?